lib/bridge/PublicitySyncer.js

```diff
@@ -181,8 +181,12 @@ PublicitySyncer.prototype._solveVisibility = Promise.coroutine(function*() {
         let currentState = this._visibilityMap.roomVisibilities[roomId];
         let correctState = shouldBePrivate(roomId, []) ? 'private' : 'public';
 
+        // Use the server domain of the first mapping
+        // 'localhost #channel1' => 'localhost'
+        const networkId = this._visibilityMap.mappings[roomId][0].split(' ')[0];
```

Comment:
Also, ignoring the null guard issue, will there always be a mapped channel for this room ID? Now that I think is no, which would then NPE on this line.